pallets/pallet-nft/src/tests.rs

```diff
@@ -0,0 +1,210 @@
+#![cfg(test)]
+
+use crate::{mock::*, *};
+use frame_support::{assert_noop, assert_ok};
+
+/// Set up an Nft to each storage which is easier to test burn, transfer and audit functionality.
+fn set_up_nfts() {
+	Nfts::<Runtime>::insert(
+		1u32,
+		NftData {
+			nft_id: 1u32,
+			data: vec![0x4E, 0x46, 0x54],
+			file_name: vec![0x46, 0x49, 0x4C, 0x45],
+		},
+	);
+	Owners::<Runtime>::insert(1u32, ALICE);
+	PendingNft::<Runtime>::insert(
+		2u32,
+		(
+			NftData {
+				nft_id: 2u32,
+				data: vec![0x4E, 0x46, 0x54],
+				file_name: vec![0x46, 0x49, 0x4C, 0x45],
+			},
+			BOB,
+		),
+	);
+	assert!(Owners::<Runtime>::contains_key(1));
+	assert!(Nfts::<Runtime>::contains_key(1));
+	assert!(PendingNft::<Runtime>::contains_key(2));
+}
+
+#[test]
+fn can_request_mint() {
+	default_test_ext().execute_with(|| {
+		let file_name = vec![0x46, 0x49, 0x4C, 0x45];
+		let data = vec![0x4E, 0x46, 0x54];
+		assert_ok!(Nft::request_mint(RuntimeOrigin::signed(ALICE), file_name.clone(), data));
+
+		assert!(PendingNft::<Runtime>::contains_key(1));
+
+		System::assert_last_event(RuntimeEvent::Nft(Event::<Runtime>::NFTPending {
+			file_name,
+			nft_id: 1u32,
+		}));
+	});
+}
+
+#[test]
+fn can_burned() {
+	default_test_ext().execute_with(|| {
+		set_up_nfts();
+		assert_ok!(Nft::burned(RuntimeOrigin::signed(ALICE), 1u32));
+
+		assert!(!Owners::<Runtime>::contains_key(1));
+		assert!(!Nfts::<Runtime>::contains_key(1));
+
+		System::assert_last_event(RuntimeEvent::Nft(Event::<Runtime>::NftBurned {
+			owner: ALICE,
+			nft_id: 1u32,
+		}));
+	});
+}
+
+#[test]
+fn can_transfer() {
+	default_test_ext().execute_with(|| {
+		set_up_nfts();
+		assert_eq!(Owners::<Runtime>::get(1), Some(ALICE));
+
+		assert_ok!(Nft::transfer(RuntimeOrigin::signed(ALICE), BOB, 1u32));
+
+		assert_eq!(Owners::<Runtime>::get(1), Some(BOB));
+		assert!(Nfts::<Runtime>::contains_key(1));
+
+		System::assert_last_event(RuntimeEvent::Nft(Event::<Runtime>::NftTransferred {
+			from: ALICE,
+			to: BOB,
+			nft_id: 1u32,
+		}));
+	});
+}
+
+#[test]
+fn can_approve_nft_pass() {
+	default_test_ext().execute_with(|| {
+		set_up_nfts();
+
```

Comment:
first ensure the nft doesn't exist in Nfts and Owners,
then approve the mint
then check that nft now exists in Nfts and Owners